superset-frontend/src/dashboard/util/replaceUndefinedByNull.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { cloneDeep } from 'lodash';
+
+function processObject(object: Object) {
+  const result = object;
+  Object.keys(result).forEach(key => {
+    if (result[key] === undefined) {
+      result[key] = null;
+    } else if (result[key] !== null && typeof result[key] === 'object') {
+      result[key] = processObject(result[key]);
+    }
+  });
+  return result;
+}
+
+export default function replaceUndefinedByNull(object: Object) {
```

Comment:
A couple tests for this would be nice

Reply:
Yes! Will do it in a follow-up PR. Thanks for the review!